### PR TITLE
Use python to run firebase testlab, do not expect recipe to know location of APK

### PR DESCRIPTION
**Pull request by dnfield**
- Updates the build rules to dump the APK into a better defined location
- Use python to run and validate the test
- Do not require CI to pass arguments about where APK lives

This will make it easier to migrate the recipe for this, and to add new APKs to test in the future.

related to https://github.com/flutter/flutter/issues/84618

### CODE REVIEW

**Comment by dnfield** [comment]
When this lands, https://flutter-review.googlesource.com/c/recipes/+/15580 can land, and then after that lands we can delete the shell scripts.

After that I'll look at the doing similar improvements for the scenarios.py recipe, and for the iOS scenario tests.

File: ci/firebase_testlab.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import argparse
+import glob
+import re
+import subprocess
+import sys
+import os
```
Comment (zanderso) [inline ci/firebase_testlab.py:12]: nit: alphabetize imports

File: ci/firebase_testlab.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+
+def CheckTimeline(results_dir):
+  du = subprocess.check_output([
+      'gsutil', 'du',
+      '%s/%s/*/game_loop_results/results_scenario_0.json' % (bucket, results_dir)
+  ]).strip()
+  if du == '0':
+    print('Failed to produce a timeline.')
+    return 1
```
Comment (zanderso) [inline ci/firebase_testlab.py]: Should this be a `sys.exit(1)`. The return is not checked below.

**Review by zanderso — APPROVED** [review]
lgtm

File: ci/firebase_testlab.py (hunk 2 above)
Comment (dnfield) [reply ci/firebase_testlab.py]: Yeah I noticed this after uploading it and it's fixed now

File: ci/firebase_testlab.py (hunk 1 above)
Comment (dnfield) [reply ci/firebase_testlab.py:12]: Done.

**Pull request merged**